src/simulations/NodeShortCircuit.jl

```diff
@@ -43,24 +42,31 @@ simulate(nsc::NodeShortCircuit, powergrid, x1, timespan)
 Simulates a [`NodeShortCircuit`](@ref)
 """
 function simulate(nsc::NodeShortCircuit, powergrid, x1, timespan)
-    sc_timespan = nsc.sc_timespan
-    @assert timespan[1] < sc_timespan[1]
-    @assert timespan[2] > sc_timespan[2]
+    @assert first(timespan) <= nsc.tspan_fault[1] "fault cannot begin in the past"
+    @assert nsc.tspan_fault[2] <= last(timespan) "fault cannot end in the future"
     nsc_powergrid = nsc(powergrid)
 
-    # Integrate to fault
-    prob1 = ODEProblem(rhs(powergrid), x1, (timespan[1], sc_timespan[1]))
-    sol1 = solve(prob1, Rodas4(autodiff=false))
+    problem = ODEProblem{true}(rhs(powergrid), x1, timespan)
 
-    # Integrate the fault state
-    x2 = find_valid_initial_condition(nsc_powergrid, sol1[end]) # Jump the state to be valid for the new system.
-    prob2 = ODEProblem(rhs(nsc_powergrid), x2, sc_timespan)
-    sol2 = solve(prob2, Rodas4(autodiff=false))
+    function errorState(integrator)
+        sol1 = integrator.sol
+        x2 = find_valid_initial_condition(nsc_powergrid, sol1[end]) # Jump the state to be valid for the new system.
+        integrator.f = rhs(nsc_powergrid)
+        integrator.u = x2
+    end
 
-    # Integrate after fault
-    x3 = find_valid_initial_condition(powergrid, sol2[end]) # Jump the state to be valid for the new system.
-    prob3 = ODEProblem(rhs(powergrid), x3, (sc_timespan[2], timespan[2]))
-    sol3 = solve(prob3, Rodas4(autodiff=false))
+    function regularState(integrator)
+        sol2 = integrator.sol
+        x3 = find_valid_initial_condition(nsc_powergrid, sol2[end]) # Jump the state to be valid for the new system.
+        integrator.f = rhs(powergrid)
+        integrator.u = x3
+    end
 
-    sol1, sol2, sol3
+    cb1 = DiscreteCallback(((u,t,integrator) -> t in nsc.tspan_fault[1]), errorState)
```

Comment:
Why is this written like that? For repeated faults? Might be worth a comment. Also, is the solver guaranteed to always find the t where the Callback is true?

Reply:
Meaning:

`t in nsc.tspan_fault[1]`

rather than

`t == nsc.tspan_fault[1]`

Reply:
Ok, understood it now. The precise timesteps are added as tstops to solve.

Reply:
I would like to merge into master then to be able to use this new nsc implementation, ok?